core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -160,21 +167,16 @@ public String toString() {
     protected final MethodMetadata metadata;
     private final Map<Integer, Expander> indexToExpander = new LinkedHashMap<Integer, Expander>();
 
-    private BuildTemplateByResolvingArgs(MethodMetadata metadata) {
+    private BuildTemplateByResolvingArgs(MethodMetadata metadata, Factory paramExpanderFactory) {
       this.metadata = metadata;
+      checkNotNull(paramExpanderFactory, "paramExpanderFactory");
       if (metadata.indexToExpanderClass().isEmpty()) {
         return;
       }
       for (Entry<Integer, Class<? extends Expander>> indexToExpanderClass : metadata
```

Comment:
I think the null signal here is different than empty. Null means
indexToExpander hasn't been processed. If empty, it has been processed, and
happens to be empty. Sorry I didn't make this clear earlier.

I wouldn't deprecate indexToExpanderClass, as the transient hint on
indexToExpander suggests something important.

All non-transient fields on MethodMetadata are relatively simple, they
don't hold values that might be hard to serialize. indexToExpander will,
which is why it is transient.